x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Node.java

```diff
@@ -94,6 +95,27 @@ public <E extends T> void forEachUp(Class<E> typeToken, Consumer<? super E> acti
         });
     }
 
+    /**
+     * Executes the action on every parent of the specified typeToken.
+     *
+     * @param typeToken the type of the node to search for
+     * @param action the action to execute for each parent of the specified typeToken
+     */
+    @SuppressWarnings("unchecked")
+    public <E extends T> E forEachParent(Class<E> typeToken, BiConsumer<E, ? super T> action) {
```

Comment:
Added this method to help determining if a full text function has an unsupported parent (like IN or IS NULL). 

It seemed to me it could be useful in the future as it's a tree traversal method. LMK if you think there's a better alternative for this, or I should confine this to the Verifier class.

Reply:
Could this be moved the `Verifier`? Just want to be cautious, `Node` is an important class, we change it only if it is necessary.

Reply:
> Could this be moved the `Verifier`? Just want to be cautious, `Node` is an important class, we change it only if it is necessary.

Don't forget to move this into `Verifier`, and keep `Node` unchanged.